test/uk/gov/hmrc/transitmovementsrouter/services/RoutingServiceSpec.scala

```diff
@@ -38,38 +38,58 @@ import scala.xml.NodeSeq
 
 class RoutingServiceSpec extends AnyFreeSpec with ScalaFutures with TestActorSystem with Matchers with MockitoSugar {
 
-  "Submitting a declaration" - new Setup {
-    "should generate a valid departure office and updated payload for a GB payload" in {
-      val serviceUnderTest = new RoutingServiceImpl(mockMessageConnectorProvider)
-      val payload          = createStream(cc015cOfficeOfDepartureGB)
-      val response         = serviceUnderTest.submitDeclaration(MovementType("Departure"), MovementId("movement-001"), MessageId("message-id-001"), payload)(hc)
-
-      whenReady(response.value, Timeout(2 seconds)) {
-        _.mustBe(Right(()))
-      }
+  "Submitting a declaration request" - new Setup {
+
+    MessageType.departureRequestValues.foreach {
+      messageType =>
+        s"${messageType.code} should generate a valid departure office and updated payload for a GB payload" in {
+          val serviceUnderTest = new RoutingServiceImpl(mockMessageConnectorProvider)
+          val payload          = createStream(messageOfficeOfDepartureGB(messageType.rootNode))
+          val response = serviceUnderTest.submitMessage(
+            MovementType("Departure"),
```

Comment:
I've come to the realisation that movement type is useless here. If we were really using it, we'd have to make this a sealed trait with two objects, because we have Departure here, and departures elsewhere -- yet we're using the X-Message-Type header for what this was meant to do, agliee stylee and all that.

I've raised CTCP-1337 to potentially remove this -- but that's not for this PR.